include/boost/geometry/algorithms/detail/disjoint/segment_box.hpp

```diff
@@ -64,10 +65,38 @@ struct disjoint_segment_box_sphere_or_spheroid
 
 public:
 
+    struct disjoint_info
+    {
+        enum type
+        {
+            intersect,
+            disjoint_no_vertex,
+            disjoint_vertex
+        };
+        disjoint_info(type t) : t_(t){}
+        operator type () const {return t_;}
```

Comment:
Where is this needed? is this class needed if you're converting it to enum? Why not just use enum instead?